e2e-test/trace/src/integrationTest/java/com/amazon/dataprepper/integration/trace/EndToEndRawSpanTest.java

```diff
@@ -165,19 +168,27 @@ private List<Map<String, Object>> getSourcesFromSearchHits(final SearchHits sear
             if (source.containsKey(TraceGroup.TRACE_GROUP_NAME_FIELD)) {
                 // Extract and replace traceGroupFields with searchHit fields to unify the representation
                 source.entrySet().removeIf(entry -> entry.getKey().startsWith("traceGroupFields"));
-                final Number statusCode = hit.field(TraceGroup.TRACE_GROUP_STATUS_CODE_FIELD).getValue();
+
+                final DocumentField statusCodeField = hit.field(TraceGroup.TRACE_GROUP_STATUS_CODE_FIELD);
+                if(statusCodeField != null)
+                    this.<Number>addNonNullField(source, hit, TraceGroup.TRACE_GROUP_STATUS_CODE_FIELD, field -> field.intValue());
+
                 // Restore trailing zeros for thousand, e.g. 2020-08-20T05:40:46.0895568Z -> 2020-08-20T05:40:46.089556800Z
-                final String endTime = Instant.parse(hit.field(TraceGroup.TRACE_GROUP_END_TIME_FIELD).getValue()).toString();
-                final Number durationInNanos = hit.field(TraceGroup.TRACE_GROUP_DURATION_IN_NANOS_FIELD).getValue();
-                source.put(TraceGroup.TRACE_GROUP_STATUS_CODE_FIELD, statusCode.intValue());
-                source.put(TraceGroup.TRACE_GROUP_END_TIME_FIELD, endTime);
-                source.put(TraceGroup.TRACE_GROUP_DURATION_IN_NANOS_FIELD, durationInNanos.longValue());
+                this.<String>addNonNullField(source, hit, TraceGroup.TRACE_GROUP_END_TIME_FIELD, field -> Instant.parse(field).toString());
+                this.<Number>addNonNullField(source, hit, TraceGroup.TRACE_GROUP_DURATION_IN_NANOS_FIELD, field -> field.longValue());
             }
             sources.add(source);
         });
         return sources;
     }
 
+    private <F> void addNonNullField(Map<String, Object> source, SearchHit hit, String fieldName, Function<F, Object> fieldToType) {
```

Comment:
If we do not include these fields, does the test verify that they exist later on? If not, there may be some validation missing at this point.

Reply:
Yes, tests verify that these exist.